### PR TITLE
py-selenium: Update to version 4.1.0

### PR DESCRIPTION
**Pull request by essandess**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.1 21C52 x86_64
Xcode 13.2.1 13C100

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-selenium/Portfile
Comment (pmetzger) [inline python/py-selenium/Portfile:26]: Are there other ports that cannot be updated that depend on the 2.7 version? If not, i think we can just drop support. Python 2 is dead.

**Review by pmetzger — COMMENTED** [review]
Just one thing before we potentially merge this (the maintainer has timed out): do we really need Python 2.7 support? I suspect not.

File: python/py-selenium/Portfile
Comment (essandess) [reply python/py-selenium/Portfile:26]: Python2 is EOL, deprecated, and unsupported but also still necessary for many projects. So that makes python2 undead.

I’d prefer not to remove python2 stuff unless necessary.

Comment (pmetzger) [reply python/py-selenium/Portfile:26]: Right, but we package things like Selenium for python 2 because other things in macports that are not yet updated for python 3 still need them. I can't find anything that depends on py-selenium, so that means we have no users of the python 2.7 version, so that means there isn't any reason to still support python 2.7.

Note that we're systematically removing all the python 2.7 things that have no dependents. Whenever we upgrade a "leaf" package to python 3, that often removes the need for remaining packages it depends on to support 2.7 and we recursively update.

Comment (essandess) [reply python/py-selenium/Portfile]: Thanks, I've removed Python 27.

**Comment by pmetzger** [comment]
@reneeotten could you give this a once over?

File: python/py-selenium/Portfile
Comment (reneeotten) [inline python/py-selenium/Portfile:53]: are you sure this explicit `destroot` phase is needed. I would think that setting/overriding the appropriate values above would be sufficient to use just use the `destroot` phase that is set by `python` PG? Looking at, for example, the `py-mkl` port that seems to be the case...

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: python/py-selenium/Portfile
Comment (reneeotten) [inline python/py-selenium/Portfile:31]: here you should/could use `${version}` instead of hardcoding the value, as also reported by `port lint`

File: python/py-selenium/Portfile
Comment (essandess) [reply python/py-selenium/Portfile]: Thanks, done.

File: python/py-selenium/Portfile
Comment (essandess) [reply python/py-selenium/Portfile]: Thanks, corrected and done.

File: python/py-selenium/Portfile
Comment (essandess) [reply python/py-selenium/Portfile]: Thanks, done.

BTW, is there some clever way of automatically getting the hashed URL of pypi distributions like this one? See https://pypi.org/project/selenium/#files → https://files.pythonhosted.org/packages/58/76/705b5c776f783d1ba7c630347463d4ae323282bbd859a8e9420c7ff79581/selenium-4.1.0-py3-none-any.whl.

**Comment by pmetzger** [comment]
@reneeotten Are we ready to merge?

File: python/py-selenium/Portfile
Comment (ryandesign) [reply python/py-selenium/Portfile]: You are not supposed to need to use these URLs that change with every version. The python portgroup is supposed to give you a working master_sites URL automatically. Why didn't it in this case?

Comment (reneeotten) [reply python/py-selenium/Portfile]: I think is has something to do with the `.whl` files and have seen this before as well but admittedly didn't spend much time to try and investigate it since it only affected a very small number of ports. I'm not sure if it can be fixed in the `python` PG but we'll not know unless someone will take a closer look - I can try to do so at some point.

**Pull request merged**
